### ISSUE TITLE
Mixed content not supported if there are child elements.

### ISSUE THREAD

**Issue by hvdp31** [FasterXML/jackson-dataformat-xml#196]
@XmlText is only supported if there are no child elements, support could be improved with some changes in XmlTokenStream.
I successfully made some changes in XmlTokenStream, it's working in my personal case, but it needs more tests.
If agreed, I could provide a patch.

Example:
Input string : `"<windSpeed units=\"kt\">27<radius>20</radius></windSpeed>"`
 "CxmlWindSpeed" class :
```
public class WindSpeed {

    public static class Radius {
        @JacksonXmlProperty(isAttribute = true)
        private String sector;
        @JacksonXmlProperty(isAttribute = true)
        private String units;
        @JacksonXmlText
        private int value;
        ..../ Getters and Setters code/....
    }
    @JacksonXmlProperty(isAttribute = true)
    private String units;
    @JacksonXmlProperty(isAttribute = true)
    private String source;
    @JacksonXmlText
    private int value;
    @JacksonXmlElementWrapper(useWrapping = false)
    private List<Radius> radius;
    ..../ Getters and Setters code/....
}
```

**Comment by cowtowncoder** [comment]
@hvdp31 Yes, a patch (via PR or stand-alone if small) would make sense, and would be very welcome!

**Comment by hvdp31** [comment]
Please find here a patch for XmlTockenStream. It does the trick in my own case and pass the tests provided with jackson-dataformat-xml.
I'm still interested in improving this part of code for other xml samples.

[XmlTokenStream.txt](https://github.com/FasterXML/jackson-dataformat-xml/files/301991/XmlTokenStream.txt)

**Comment by cowtowncoder** [comment]
@hvdp31 That certainly looks delightfully simple! I'll try to have a look at this soon; I just released 2.8.0.rc1, and this would make it in rc2 (in theory perhaps even backport for 2.7.5, but this is where I don't know how good our unit test coverage is).

**Comment by cowtowncoder** [comment]
Ok added implementation along suggested lines, and added a simple test. I assume handling would not work with certain other combinations (like between end- and start-tags), but supported case seems like most common one so it should be good start, or perhaps sufficient all around.
